service/src/main/java/io/camunda/service/ProcessDefinitionServices.java

```diff
@@ -39,4 +42,33 @@ public ProcessDefinitionServices withAuthentication(final Authentication authent
     return new ProcessDefinitionServices(
         brokerClient, processDefinitionSearchClient, authentication);
   }
+
+  public ProcessDefinitionEntity getByKey(final Long processDefinitionKey) {
+    final SearchQueryResult<ProcessDefinitionEntity> result =
+        search(
+            SearchQueryBuilders.processDefinitionSearchQuery()
+                .filter(f -> f.processDefinitionKeys(processDefinitionKey))
+                .build());
+    if (result.total() < 1) {
+      throw new NotFoundException(
+          String.format("Process Definition with key %d not found", processDefinitionKey));
+    } else if (result.total() > 1) {
+      throw new CamundaSearchException(
+          String.format(
+              "Found Process Definition with key %d more than once", processDefinitionKey));
+    } else {
+      return result.items().stream().findFirst().orElseThrow();
+    }
+  }
+
+  public String getProcessDefinitionXml(final Long processDefinitionKey) {
+    final var processDefinition = getByKey(processDefinitionKey);
+    final var xml = processDefinition.bpmnXml();
+    if (xml == null) {
```

Comment:
❓ Should we return 204 here? With No Content? 
I would leave Not Found (404) For in case the process-definition is not found

Over this case, I see the process was found, but there is no content to present on XML

Reply:
Good point. I've oriented at the behaviour of [Get decision definition XML](https://github.com/camunda/camunda/blob/c9f58b26017bfc388e6ccfab35b56c372e833252/service/src/main/java/io/camunda/service/DecisionDefinitionServices.java#L77-L77).